From d061c1a47907aa3a59de44143ca2479d4ef7c130 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:13 -0400
Subject: [cpufreq] Use signed type for 'ret' variable, to store negative error values

Message-id: <1409235879-1283-40-git-send-email-prarit@redhat.com>
Patchwork-id: 89377
O-Subject: [RHEL7.1 PATCH BZ 1134639 039/185] cpufreq: Use signed type for 'ret' variable, to store negative error values
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 5136fa56582beadb7fa71eb30bc79148bfcba5c1
Author: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
Date:   Sat Sep 7 01:24:06 2013 +0530

    cpufreq: Use signed type for 'ret' variable, to store negative error values

    There are places where the variable 'ret' is declared as unsigned int
    and then used to store negative return values such as -EINVAL. Fix them
    by declaring the variable as a signed quantity.

    Signed-off-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index fcfbccc..e643c60 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -437,7 +437,7 @@ static int __cpufreq_set_policy(struct cpufreq_policy *policy,
 static ssize_t store_##file_name     \
 (struct cpufreq_policy *policy, const char *buf, size_t count)  \
 {         \
- unsigned int ret;      \
+ int ret;       \
  struct cpufreq_policy new_policy;    \
          \
  ret = cpufreq_get_policy(&new_policy, policy->cpu);  \
@@ -490,7 +490,7 @@ static ssize_t show_scaling_governor(struct cpufreq_policy *policy, char *buf)
 static ssize_t store_scaling_governor(struct cpufreq_policy *policy,
      const char *buf, size_t count)
 {
- unsigned int ret;
+ int ret;
  char str_governor[16];
  struct cpufreq_policy new_policy;
 
-- 
1.7.1